### PR TITLE
Switch to a producer/consumer queue to search for computing refactorings

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs [hunk 1]
```diff
@@ -119,44 +119,55 @@ static ProjectCodeRefactoringProvider.ExtensionInfo GetExtensionInfo(ExportCodeR
...
+                            {
+                                m[TelemetryLogging.KeyName] = providerName;
+                                m[TelemetryLogging.KeyLanguageName] = args.document.Project.Language;
+                            });
+
+                            using (args.addOperationScope(providerName))
+                            using (RoslynEventSource.LogInformationalBlock(FunctionId.Refactoring_CodeRefactoringService_GetRefactoringsAsync, providerName, cancellationToken))
+                            using (TelemetryLogging.LogBlockTime(FunctionId.CodeRefactoring_Delay, logMessage, CodeRefactoringTelemetryDelay))
+                            {
+                                var extensionManager = args.document.Project.Solution.Services.GetRequiredService<IExtensionManager>();
```
Comment (ToddGrun) [inline src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:154]: >var extensionManager [](http://example.com/codeflow?start=32&length=20)

out of curiosity, why was this moved inside the producer code? Couldn't it have been passed in like it was before?

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:154]: very true.  it was more annoyance around passing the args along :)

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski For review when you get back.
